pkg/network/http/driver_interface.go

```diff
@@ -0,0 +1,238 @@
+	return d, nil
+}
+
+func (di *httpDriverInterface) setupHTTPHandle() error {
+	dh, err := driver.NewHandle(windows.FILE_FLAG_OVERLAPPED, driver.HTTPHandle)
+	if err != nil {
+		return err
+	}
+
+	filters, err := createHTTPFilters()
+	if err != nil {
+		return err
+	}
+
+	if err := dh.SetHTTPFilters(filters); err != nil {
+		return err
+	}
+
+	iocp, buffers, err := driver.PrepareCompletionBuffers(dh.Handle, httpReadBufferCount)
+	if err != nil {
+		return err
+	}
+
+	di.driverHTTPHandle = dh
+	di.iocp = iocp
+	di.readBuffers = buffers
+	return nil
+}
+
+func createHTTPFilters() ([]driver.FilterDefinition, error) {
+	ifaces, err := net.Interfaces()
+	if err != nil {
+		return nil, err
+	}
+
+	var filters []driver.FilterDefinition
+	for _, iface := range ifaces {
+		// IPv4
+		filters = append(filters, driver.FilterDefinition{
+			FilterVersion:  driver.Signature,
+			Size:           driver.FilterDefinitionSize,
+			Direction:      driver.DirectionOutbound,
+			FilterLayer:    driver.LayerTransport,
+			InterfaceIndex: uint64(iface.Index),
+			Af:             windows.AF_INET,
+			Protocol:       windows.IPPROTO_TCP,
+		}, driver.FilterDefinition{
+			FilterVersion:  driver.Signature,
+			Size:           driver.FilterDefinitionSize,
+			Direction:      driver.DirectionInbound,
+			FilterLayer:    driver.LayerTransport,
+			InterfaceIndex: uint64(iface.Index),
+			Af:             windows.AF_INET,
+			Protocol:       windows.IPPROTO_TCP,
+		})
+
+		// IPv6
+		filters = append(filters, driver.FilterDefinition{
+			FilterVersion:  driver.Signature,
+			Size:           driver.FilterDefinitionSize,
+			Direction:      driver.DirectionOutbound,
+			FilterLayer:    driver.LayerTransport,
+			InterfaceIndex: uint64(iface.Index),
+			Af:             windows.AF_INET6,
+			Protocol:       windows.IPPROTO_TCP,
+		}, driver.FilterDefinition{
+			FilterVersion:  driver.Signature,
+			Size:           driver.FilterDefinitionSize,
+			Direction:      driver.DirectionInbound,
+			FilterLayer:    driver.LayerTransport,
+			InterfaceIndex: uint64(iface.Index),
+			Af:             windows.AF_INET6,
+			Protocol:       windows.IPPROTO_TCP,
+		})
+	}
+
+	return filters, nil
+}
+
+func (di *httpDriverInterface) setMaxFlows(maxFlows uint64) error {
+	log.Debugf("Setting max flows in driver http filter to %v", maxFlows)
+	err := windows.DeviceIoControl(di.driverHTTPHandle.Handle,
+		driver.SetMaxFlowsIOCTL,
+		(*byte)(unsafe.Pointer(&maxFlows)),
+		uint32(unsafe.Sizeof(maxFlows)),
+		nil,
+		uint32(0), nil, nil)
+	if err != nil {
+		log.Warnf("Failed to set max number of flows in driver http filter to %v %v", maxFlows, err)
+	}
+	return err
+}
+
+func (di *httpDriverInterface) startReadingBuffers() {
+	di.eventLoopWG.Add(1)
+	go func() {
+		defer di.eventLoopWG.Done()
+
+		transactionSize := uint32(C.size_of_http_transaction_type())
+		for {
+			buf, bytesRead, err := driver.GetReadBufferWhenReady(di.iocp)
+			if iocpIsClosedError(err) {
+				return
```

Comment:
Does this loop exiting cause the error to be surfaced elsewhere, or would we silently stop monitoring HTTP?

Reply:
I'm not returning/surfacing this error because it's the intended exit condition for this loop - the IOCP won't close until we close the file handle which we've associated with it, so this should only occur when we call `httpDriverInterface.close`. I'll add a debug log line here so that the process isn't silent though